src/main/java/fr/ens/biologie/genomique/eoulsan/modules/diffana/DESeq2Checker.java

```diff
@@ -121,34 +158,53 @@ static boolean checkExperimentDesign(Experiment experiment, boolean throwsExcept
         // If mode is not complex
         if (!emd.containsComparisons()) {
             // If Exp.exp1.Condition and Exp.exp1.Reference exist
-            if (getExperimentSampleAllMetadataKeys(experiment).contains(REFERENCE_KEY) && getExperimentSampleAllMetadataKeys(experiment).contains(CONDITION_KEY)){
+            if (getExperimentSampleAllMetadataKeys(experiment).contains(REFERENCE_KEY)
+                    && getExperimentSampleAllMetadataKeys(experiment).contains(CONDITION_KEY)){
                 Map<String, String> lhm = new LinkedHashMap<>();
-                // Get all values of the Condition and Reference columns
+                // Get all samples values of the Condition and Reference columns
                 for (ExperimentSample es : experiment.getExperimentSamples()) {
                     String conditions = DesignUtils.getMetadata(es, CONDITION_KEY);
                     String references = DesignUtils.getMetadata(es, REFERENCE_KEY);
                     // If one condition is associated with more than one reference, error
-                    ArrayList<String> possibleConditionsReferences = new ArrayList<String>();
+                    List<String> possibleConditionsReferences = new ArrayList<>();
                     possibleConditionsReferences.add(conditions+references);
-                    // if condition already in dictionnary check if value is equals to reference and if not error
                     for (String key : lhm.keySet()) {
                         String value = lhm.get(key);
                         if (key.equals(conditions) && !value.equals(references)){
-                            return error("There is an inconsistency between the conditions and the references : "
-                                    + experiment.getName(), throwsException);
+                            return error("There is an inconsistency between the conditions " +
+                                    "and the references : " + experiment.getName(), throwsException);
                         }
                     }
                     lhm.put(conditions,references);
                 }
             }
         }
 
-        // Verify consistency between "Exp.exp1" in the header of the file and the ExperimentSample columns name
-        // TO DO
+        /* Check if there is no combination column-name that is equal to another column-name or to a column name */
+        // Get every sample value for each key, and get every possible combination by merging the strings
+        ArrayList<String> possibleCombination = new ArrayList<String>();
+        for (String key : allkeys) {
+            for (ExperimentSample es : experiment.getExperimentSamples()) {
+                String value = DesignUtils.getMetadata(es, key);
+                String possibleCombinationvalueone = key+value;
+                String possibleCombinationvaluetwo = value+key;
+                possibleCombination.add(possibleCombinationvalueone);
```

Comment:
You must use a multimap to check if for two different key and value, value+key can be equals.